lib/optimize/CommonsChunkPlugin.js

```diff
@@ -24,153 +24,304 @@ The available options are:
 	minSize: number`);
 		}
 
-		if(Array.isArray(options) || typeof options === "string") {
-			options = {
-				name: options
+		const normalizedOptions = this.normalizeOptions(options);
+
+		this.chunkNames = normalizedOptions.chunkNames;
+		this.filenameTemplate = this.filenameTemplate;
+		this.minChunks = normalizedOptions.minChunks;
+		this.selectedChunks = normalizedOptions.selectedChunks;
+		this.async = normalizedOptions.async;
+		this.minSize = normalizedOptions.minSize;
+		this.ident = normalizedOptions.ident;
+	}
+
+	normalizeOptions(options) {
+		if(Array.isArray(options)) {
+			return {
+				chunkNames: options,
+			};
+		}
+
+		if(typeof options === "string") {
+			return {
+				chunkNames: [options],
 			};
 		}
-		this.chunkNames = options.name || options.names;
-		this.filenameTemplate = options.filename;
-		this.minChunks = options.minChunks;
-		this.selectedChunks = options.chunks;
-		if(options.children) this.selectedChunks = false;
-		this.async = options.async;
-		this.minSize = options.minSize;
-		this.ident = __filename + (nextIdent++);
+
+		// if "children" is set, set selectedChunks to false instead of specified chunks
+		// TODO: why? :P
+		const selectedChunks = options.children ? false : options.chunks;
+		const chunkNames = options.name ? [options.name] : options.names;
```

Comment:
why is that?
in the current implementation the options are either a string or an array of strings:
https://github.com/webpack/webpack/blob/master/lib/optimize/CommonsChunkPlugin.js#L32
and then "become" an array here:
https://github.com/webpack/webpack/blob/master/lib/optimize/CommonsChunkPlugin.js#L58-L59